kernel/aster-nix/src/net/socket/unix/stream/listener.rs

```diff
@@ -159,11 +160,10 @@ impl Backlog {
     }
 
     fn push_incoming(&self, endpoint: Endpoint) -> Result<()> {
-        let mut endpoints = self.incoming_endpoints.lock();
-        if endpoints.len() >= self.backlog {
-            return_errno_with_message!(Errno::ECONNREFUSED, "incoming_endpoints is full");
+        while self.incoming_endpoints.lock().len() >= self.backlog {
+            Thread::yield_now();
```

Comment:
Blocking behavior should be implemented in `socket.rs` to avoid holding locks while blocking.

Also, it looks like we need to add more `Pollee` to avoid this `Thread::yield_now()`. The current `Pollee` is not suitable for reuse because it contains I/O events for the user, but here we need to notify the connecting sockets, which is an implementation detail for the kernel, so it's I/O events inside the kernel.

I'm doing refactoring work on UNIX sockets, but actually I didn't notice this behavioral difference to the Linux kernel until I saw this PR. Give me some time to think more about it.

Reply:
> The current Pollee is not suitable for reuse because it contains I/O events for the user, but here we need to notify the connecting sockets, which is an implementation detail for the kernel, so it's I/O events inside the kernel.

Yes, I think that is a general problem.